session.go

```diff
@@ -114,6 +161,112 @@ func NewSession(cfg ClusterConfig) (*Session, error) {
 		return nil, errors.New("Can't use both Authenticator and AuthProvider in cluster config.")
 	}
 
+	// Needs to be here because we need to fill in hosts and sniconfig before we do anything
+	var sniConfig *SNIConfig
+
+	if cfg.SecureConnectBundleFilename != "" {
+		// 1. Unzip the secure bundle.
+		r, err := zip.OpenReader(cfg.SecureConnectBundleFilename)
+		if err != nil {
+			return nil, err
+		}
+		defer r.Close()
+
+		// 2. Put in temp directory (the following MUST BE FILES for this to work.
+		//    a) 'cert' file
+		//    b) 'key' file
+		//    c) 'ca.crt' file
+
+		dir, err := ioutil.TempDir("", "securezip")
+		if err != nil {
+			return nil, err
+		}
+		defer os.RemoveAll(dir) // the files are only needed until we create the tlsConfig, at that point they have been read in and processed, so not needed any longer and can be deleted at end of method.
+
+		// add each file from bundle to the directory
+		for _, f := range r.File {
+			fpath := filepath.Join(dir, f.Name)
```

Comment:
Extracting a file like this is a security issue - if the zip was malicious, it could replace any file it wanted. The documentation for `Name` says:

```
    // It is the caller's responsibility to sanitize it as
    // appropriate, including canonicalizing slash directions,
    // validating that paths are relative, and preventing path
    // traversal through filenames ("../../../").
```